synapse/daemon.py

```diff
@@ -378,10 +378,11 @@ def _loadDmonConf(self):
                 items = list(link.get('dmon:items').values())
 
                 for item in items:
-                    try:
-                        await item.fini()
-                    except Exception as e:
-                        logger.exception(f'item fini error: {e}')
+                    if hasattr(item, 'fini'):
```

Comment:
This is problematic if the item inherits from EventBus (and therefore has a non-async `fini`).  Perhaps check if inherits from s_coro.Fini instead?

Reply:
Thinking about this further, I'm actually not convinced that fini'ing an object when the share link associated with it is fini'd if we are already fini'ing objects when the daemon itself is torn down.

Reply:
+1 on checking for isinstance(s_coro.Fini) and it is needed for dynamically shared objects which are only expected to survive across the socket ( and would otherwise leak forever )